src/lang/ts/intermock.ts

```diff
@@ -321,6 +346,78 @@ function processArrayPropertyType(
   }
 }
 
+/**
+ * Process an array definition.
+ *
+ * @param node Node being processed
+ * @param output The object outputted by Intermock after all types are mocked
+ * @param property Output property to write to
+ * @param typeName Type name of property
+ * @param kind TS data type of property type
+ * @param sourceFile TypeScript AST object compiled from file data
+ * @param options Intermock general options object
+ * @param types Top-level types of interfaces/aliases etc.
+ */
+function processUnionPropertyType(
+    node: ts.PropertySignature, output: Output, property: string,
+    typeName: string, kind: ts.SyntaxKind, sourceFile: ts.SourceFile,
+    options: Options, types: Types) {
+  // @ts-ignore
+  const unionNodes = node && node.type ?
+      (node.type as ts.UnionTypeNode).types as ts.NodeArray<ts.TypeNode>:
+      [];
+  // @ts-ignore
+  const supportedType = unionNodes.find(
+      type =>
+          Object.keys(SupportedTypes)
+              .map(key => (SupportedTypes as unknown as {[key: string]:
```

Comment:
You can change `SupportedTypes` from an `enum` to an interface and add an index signature to get around casting to unknown and then to a distinct type.

Reply:
Done